Makefile

```diff
@@ -6,206 +6,211 @@
 
 HELM_NAMESPACE := instill-ai
-HELM_RELEASE_NAME := vdp
+HELM_RELEASE_NAME := instill-ai
 
 #============================================================================
 
 .PHONY: all
 all:			## Launch all services with their up-to-date release version
-	@if [ "${BUILD}" = "true" ]; then make build-release; fi
-	@if [ ! "$$(docker image inspect ${CONTAINER_COMPOSE_IMAGE_NAME}:${INSTILL_VDP_VERSION} --format='yes' 2> /dev/null)" = "yes" ]; then \
+	@make build-release BUILD=${BUILD}
+	@if [ ! -f "$$(echo ${SYSTEM_CONFIG_PATH}/user_uid)" ]; then \
+		mkdir -p ${SYSTEM_CONFIG_PATH} && \
+		docker run --rm --name uuidgen ${INSTILL_CORE_IMAGE_NAME}:${INSTILL_CORE_VERSION} uuidgen > ${SYSTEM_CONFIG_PATH}/user_uid; \
+	fi
+ifeq (${NVIDIA_GPU_AVAILABLE}, true)
+	@docker inspect --type=image instill/ray:${RAY_SERVER_VERSION} >/dev/null 2>&1 || printf "\033[1;33mINFO:\033[0m This may take a while due to the enormous size of the Ray server image, but the image pulling process should be just a one-time effort.\n" && sleep 5
+	@cat docker-compose-nvidia.yml | yq '.services.ray_server.deploy.resources.reservations.devices[0].device_ids |= (strenv(NVIDIA_VISIBLE_DEVICES) | split(",")) | ..style="double"' | \
+		EDITION=$${EDITION:=local-ce} DEFAULT_USER_UID=$$(cat ${SYSTEM_CONFIG_PATH}/user_uid) docker compose ${COMPOSE_FILES} -f - up -d --quiet-pull
+else
+	@EDITION=$${EDITION:=local-ce} DEFAULT_USER_UID=$$(cat ${SYSTEM_CONFIG_PATH}/user_uid) docker compose ${COMPOSE_FILES} up -d --quiet-pull
+endif
+
+.PHONY: latest
+latest:			## Lunch all dependent services with their latest codebase
+	@make build-latest PROFILE=${PROFILE}
+	@if [ ! -f "$$(echo ${SYSTEM_CONFIG_PATH}/user_uid)" ]; then \
+		mkdir -p ${SYSTEM_CONFIG_PATH} && \
+		docker run --rm --name uuidgen ${INSTILL_CORE_IMAGE_NAME}:latest uuidgen > ${SYSTEM_CONFIG_PATH}/user_uid; \
+	fi
+ifeq (${NVIDIA_GPU_AVAILABLE}, true)
+	@docker inspect --type=image instill/ray:latest-${RAY_PLATFORM} >/dev/null 2>&1 || printf "\033[1;33mINFO:\033[0m This may take a while due to the enormous size of the Ray server image, but the image pulling process should be just a one-time effort.\n" && sleep 5
+	@cat docker-compose-nvidia.yml | yq '.services.ray_server.deploy.resources.reservations.devices[0].device_ids |= (strenv(NVIDIA_VISIBLE_DEVICES) | split(",")) | ..style="double"' | \
+		COMPOSE_PROFILES=${PROFILE} EDITION=$${EDITION:=local-ce:latest} DEFAULT_USER_UID=$$(cat ${SYSTEM_CONFIG_PATH}/user_uid) docker compose ${COMPOSE_FILES} -f docker-compose-latest.yml -f - up -d --quiet-pull
+else
+	@COMPOSE_PROFILES=${PROFILE} EDITION=$${EDITION:=local-ce:latest} DEFAULT_USER_UID=$$(cat ${SYSTEM_CONFIG_PATH}/user_uid) docker compose ${COMPOSE_FILES} -f docker-compose-latest.yml up -d --quiet-pull
+endif
+
+.PHONY: build-latest
+build-latest:				## Build latest images for all VDP components
+	@if [ ! "$$(docker image inspect ${INSTILL_CORE_IMAGE_NAME}:latest --format='yes' 2> /dev/null)" = "yes" ]; then \
```

Comment:
I suggest we don't need to add the if statement here because, after the first build, we'll always have this image, and it will no longer be built again.